API/projectClass/backgroundsClass.m

```diff
@@ -125,7 +134,7 @@
             % Changes the value of a given background in the table. Expects the 
             % index or name of background and keyword/value pairs to set. 
             %
-            % background.setBackground(1, 'name', 'back 1', 'type', 'constant', 'value1', 'param_name');
+            % background.setBackground(1, 'name', 'back 1', 'type', 'constant', 'source', 'param_name');
             if isText(row)
```

Comment:
I think this function  `setBackground` needs to be refactored  because it is possible to change the type without updating the source also when I change the **source**, **value 1** to **value 5** remains the same which can have unexpected consequences for example if I switch from **function** to **data**, **value 1** now becomes the unwilling offset for my data and this will not be caught by any checks

Reply:
I see two options here: either bar the user from changing the background type, or have the result of changing the type being that the source and all values are cleared. I think the second of these is not intuitive to users, so we should take the first option and make sure the background type is not changable. This will also be easy to transfer to the python, since changing the type there without updating anything else will trigger validation errors in the source field.

Reply:
I think barring the user from changing the type is the confusing option. You mean for type alone the user will have to delete the background and add a new one. I would prefer the source and values1-5 are reset if the user changes the type without setting them and a warning is printed to notify the user what and why

Reply:
I see what you mean, but in ALL cases of changing the type, the source field must also change. Changing the type and the source is not far off defining a new background anyway. Also, the old background does not have to be deleted, it can be retained and switched in the contrast(s). I can see the user case of having a data background and a function background and wanting to try out each one.

Reply:
My concern as always is consistency the set function should set stuff not direct the user to the add function to add a new background even if the user has to set most of the fields. If the user wants to alternate between types they will use the add function instead.

Reply:
I get your point, but setting one field and clearing six others isn't great for consistency either. Hmmm.

Reply:
So what is your solution for telling the user they cannot change the type  field? throw an exception?

Reply:
Yeah, I think so. We raise an exception for trying to remove a protected parameter, so it falls in that area I guess. I'll have to have a think about that one.

Reply:
We do not throw any exception anywhere for changing a field, we throw a warning and reset the data range, we throw a warning an reset the prior mu and sigma. The reason for not deleting a protected parameter is well understood by the user

Reply:
If you change the min/max/value of a parameter range such that: "NOT (min<=value<=max)", an exception is thrown.